#load the RNA-seq data set
#set your working directory and load the data
setwd("~/Desktop/R/ComputationalGenomicsR/")
df <- read.table('vehicle_drug_feature_counts.txt',
                 header = T, sep = '\t', row.names = 1)
summary(df)

df <- df[,6:9]

#save the data
write.table(df, 
            file = 'vehicle_drug_feature_counts.copy.txt.txt', #either a character string naming a file or a connection open for writing. "" indicates output to the console.
            row.names = T, #indicating whether the row names of df are to be written
            col.names = T, #indicating whether the column names of df are to be written
            sep = '\t',    #the field separator string
            quote = F      #a logical value (TRUE or FALSE) or a numeric vector. If TRUE, any character or factor columns will be surrounded by double quotes.
)

###load again
df2 <- read.table('vehicle_drug_feature_counts.copy.txt.txt',header = T, sep = '\t', row.names = 1)
head(df2)




#for loop
df <- as.matrix(df)

##check the quantile for each assay
for (i in 1:ncol(df))
{
  print(  colnames(df)[i] )
  qu <- quantile( df[, i] )
  print(qu)
}


#apply replaces for loop
qu <- apply(df, 2, function(z){quantile(z)})
print(qu)





#if ... else statement

### pick the expressed genes in first assay
exp_cutoff   <- quantile(df[,1], 0.35)
unEXP_cutoff <- quantile(df[,1], 0.3)
n_remaining  <- 0
exp_gene   <- c() #set a empty vector for expressed genes
unexp_gene <- c() #set a empty vector for un_expressed genes
for (i in (1:length(df[,1])))
{
  if (df[i, 1] > exp_cutoff  )
  {
    exp_gene <- c(exp_gene, rownames(df)[i])  
  }
  else if (df[i, 1] < unEXP_cutoff) 
  {
    unexp_gene <- c(unexp_gene, rownames(df)[i])
  }
  else
  {
    n_remaining <- n_remaining + 1    
  }
}

print(c(length(unexp_gene),
        length(exp_gene),
        n_remaining))


#use 'which' to replace former script
exp_gene2 <- rownames(df)[which(df[,1] > exp_cutoff)]
length(exp_gene2)

length(exp_gene)

